src/hotspot/share/classfile/klassFactory.cpp

```diff
@@ -76,7 +76,7 @@ InstanceKlass* KlassFactory::check_shared_class_file_load_hook(
       s2 path_index = ik->shared_classpath_index();
       ClassFileStream* stream = new ClassFileStream(ptr,
                                                     pointer_delta_as_int(end_ptr, ptr),
-                                                    cfs->source());
+                                                    cfs->source(), false, /* from_class_file_load_hook */ true);
```

Comment:
Can you comment what the `false` means as well please - looks odd to only document one of the boolean parameters.